projects/aca-content/src/lib/services/modal-ai.service.spec.ts

```diff
@@ -0,0 +1,112 @@
+/*!
+ * Copyright © 2005-2024 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ModalAiService } from './modal-ai.service';
+import { TestBed } from '@angular/core/testing';
+import { ContentTestingModule } from '@alfresco/adf-content-services';
+import { MatDialog, MatDialogModule, MatDialogRef } from '@angular/material/dialog';
+import { ActivatedRoute, Params } from '@angular/router';
+import { of, Subject } from 'rxjs';
+import { StorageService, UnsavedChangesDialogComponent } from '@alfresco/adf-core';
+
+describe('ModalAiService', () => {
+  const mockQueryParams = new Subject<Params>();
+  let service: ModalAiService;
+  let dialogOpenSpy: jasmine.Spy<(component: any, config?: any) => MatDialogRef<any>>;
```

Comment:
We should avoid any. You can change it to: 
let dialogOpenSpy: jasmine.Spy<<T, D, R>(component: ComponentType<T>, config?: MatDialogConfig<D>) => MatDialogRef<T, R>>;

or any other if it avoids any.